Source/RapyutaSimulationPlugins/Public/Core/RRGeneralUtils.h

```diff
@@ -294,4 +294,56 @@ class RAPYUTASIMULATIONPLUGINS_API URRGeneralUtils : public UBlueprintFunctionLi
         OutValue = InDefaultValue;
         return false;
     }
+
+    /**
+     * @brief Check if a velocity's magnitude exceeds a given max speed
+     *
+     * @param InVel
+     * @param InMaxSpeed
+     * @param b2DMovement
+     */
+    FORCEINLINE static bool IsVelocityExceedingMaxSpeed(const FVector& InVel, float InMaxSpeed, bool b2DMovement)
+    {
+        // Give 1% error tolerance, to account for numeric imprecision
+        static constexpr float OVER_VEL_PERCENT = 1.01f;
+        const float maxExtraSpeed = OVER_VEL_PERCENT * FMath::Square(InMaxSpeed);
+        return (maxExtraSpeed > 0.f) &&
+               (b2DMovement ? (InVel.SizeSquared2D() > maxExtraSpeed) : (InVel.SizeSquared() > maxExtraSpeed));
+    }
+
+    /**
+     * @brief Set a given velocity's magnitude to its max allowed speed
+     *
+     * @param InVel
+     * @param InMaxSpeed
+     * @param b2DMovement
+     */
+    FORCEINLINE static void SetVelocityToMaxSpeed(FVector& InVel, float InMaxSpeed, bool b2DMovement)
```

Comment:
Can we rename to Vector instead of Velocity?

Reply:
Same reason as above but then since Velocity is always a 3d-vector only, and basically we also want to provide `b2DMovement` as Velocity-specific param